check-daily-slack.py

```diff
@@ -40,21 +40,29 @@ def debug(msg):
     return
   print("%s> %s" % (datetime.isoformat(datetime.utcnow()), msg))
 
+def is_on_cvmfs(pkg, tag):
+  if os.environ.get("CVMFS_CHECK_POSIX", "0").lower() in [ "1", "yes", "true", "on" ]:
+    return is_on_cvmfs_posix(pkg, tag)
+  return is_on_cvmfs_http(pkg, tag)
+
 @quench
-def is_on_cvmfs_parrot(pkg, tag):
-  # Check if found on CVMFS on at least one architecture (use parrot)
-  parrot = [ "env",
-             "HTTP_PROXY=DIRECT;",
-             "PARROT_ALLOW_SWITCHING_CVMFS_REPOSITORIES=yes",
-             "PARROT_CVMFS_REPO=<default-repositories>",
-             "parrot_run" ]
-  for d in getout(parrot + ["ls", "-1", "/cvmfs/alice.cern.ch"])[0].split("\n"):
-    if getout(parrot + ["bash", "-c", "ls -1 /cvmfs/alice.cern.ch/%s/Modules/modulefiles/%s/%s*" % (d, pkg, ver)])[1] == 0:
-      return True
+def is_on_cvmfs_posix(pkg, tag):
+  # Check if found on CVMFS on at least one architecture
+  parrot = []
+  for _ in range(0,2):
+    for d in getout(parrot + ["ls", "-1", "/cvmfs/alice.cern.ch"])[0].split("\n"):
+      if getout(parrot + ["bash", "-c", "ls -1 /cvmfs/alice.cern.ch/%s/Modules/modulefiles/%s/%s*" % (d, pkg, ver)])[1] == 0:
+        return True
+    # Fallback on Parrot next time
+    parrot = [ "env",
+               "HTTP_PROXY=DIRECT;",
+               "PARROT_ALLOW_SWITCHING_CVMFS_REPOSITORIES=yes",
+               "PARROT_CVMFS_REPO=<default-repositories>",
+               "parrot_run" ]
   return False
 
```

Comment:
Can this not be simplified?

```python
def is_on_cvmfs_posix(pkg, tag):
   for parrot in ([], ["env", .....]):
      if getout(parrot + ["bash", "-c", "ls -1 /cvmfs/alice.cern.ch/*/Modules/....."])[1] == 0:
         return True
   return False
```

Reply:
Well, the `for` line would become incredibly long. This is why I did not do it that way.

Reply:
Well, you could always put them into a variable, broken over several lines,
as in your version and then for-loop over the variable.
But anyhow, this is not very important so let me not hold you up! 😉 
It was just that when I looked at the code I was confused for some seconds
wondering what the range loop was all about.

Reply:
Your solution is more Pythonic (`([], long_variable)`), so I am going for it. I did not like the `range()` either to be honest, this was a dirty Christmas fix!